src/server.c

```diff
@@ -680,6 +680,52 @@ int clientsCronResizeQueryBuffer(client *c) {
     return 0;
 }
 
+/* The client output buffer can be adjusted to better fit the memory requierments.
+ *
+ * the logic is:
+ * in case the last observed peak size of the buffer equals the buffer size - we double the size
+ * in case the last observed peak size of the buffer is less than half the buffer size - we shrink by half.
+ * The buffer peak will be reset back to zero every 5 seconds
+ * The function always returns 0 as it never terminates the client. */
+int clientsCronResizeOutputBuffer(client *c) {
+
+    size_t new_buffer_size = 0;
+    const size_t buffer_target_shrink_size = c->buf_usable_size/2;
+    const size_t buffer_target_expend_size = c->buf_usable_size*2;
+
+    if (buffer_target_shrink_size >= PROTO_REPLY_MIN_BYTES &&
+        c->buf_peak < buffer_target_shrink_size ) {
+        new_buffer_size = buffer_target_shrink_size;
```

Comment:
styling: `if` condition with multiple lines, should have the open `{` in a separate line.
otherwise, it's hard to distinguish between the condition and the actions.

p.s. the alignment of the additional condition lines should probably either be indented by 4 compared to outer scope, or indented to after the `if` opening `(`.
see the indentation of the `else-if` below.

p.p.s that `else-if` is also missing a space before the `(`